code/modules/mob/living/simple_animal/hostile/megafauna/megafauna.dm

```diff
@@ -132,6 +132,9 @@
 	recovery_time = world.time + buffer_time
 	ranged_cooldown = world.time + buffer_time
 
+/mob/living/simple_animal/hostile/megafauna/proc/can_act()
+	return !QDELETED(src) && !(stat == DEAD) //just in case admins delete it or something and it's in GC queue and not dead
```

Comment:
couldn't you just make this as a define instead of proc? Things that would change if a megafauna could attack like watcher wing are put on the object rather than checked here so we wouldn't need to overwrite this.

`#define MOB_ALIVE_AND_WELL(X)>> (!QDELETED(X) && (X.stat != DEAD))`

Reply:
sure